OMCompiler/SimulationRuntime/c/simulation/solver/kinsol_b.c

```diff
@@ -1658,6 +1727,11 @@ NLS_SOLVER_STATUS B_nlsKinsolSolve(DATA* data, threadData_t* threadData, NONLINE
       svd_compute(data, nlsData, SM_DATA_S(kinsolData->J), kinsolData->useScaling, KINSOL_B_ENTRY_POINT /* called at entry point */);
     }
 
+    const char *arg = omc_flagValue[FLAG_SAVE_INITIAL_GUESS_SYSTEM];
+    if (arg) {
+      B_save_initial_guess_system(data, threadData, nlsData, arg);
```

Comment:
```suggestion
    if (omc_flag[FLAG_SAVE_INITIAL_GUESS_SYSTEM]) {
      B_save_initial_guess_system(data, threadData, nlsData, omc_flagValue[FLAG_SAVE_INITIAL_GUESS_SYSTEM]);
```
And you don't even need to pass the flag value into the function as you can use it directly there.